### ISSUE TITLE
Training going extremely slow.

### ISSUE THREAD

**Issue by TomasGuija** [thuliu-yt16/dualvector#6]
I was trying to train a model as you explain on your Readme, in this case for the font reconstruction task. However it is going extremely slow. The call to loss.backward() is taking an average of around 30s, making each epoch take too long. I've been wandering through the code trying to figure out what could be provoking this issue for a couple of days but I couldn't find it. Do you have any guess, or could you have experienced something similar? After profiling, I found that the bottleneck may be due to some scalar extraction (.item() calls). Here is what the Pytorch profiler reports: 

aten::_local_scalar_dense    CPU total: 41.575 s (84.47%)
aten::item                   CPU total: 41.604 s (84.53%)

 Thank you very much for your time.

**Comment by thuliu-yt16** [comment]
That is weird. I have never experienced such problem. What GPU did you use for training?

**Comment by thuliu-yt16** [comment]
The profiler did not provide much information. Could you try to simply profile/time the program with CUDA_LAUNCH_BLOCKING=1？
